src/Propel/Runtime/ActiveQuery/ModelCriteria.php

```diff
@@ -440,29 +434,36 @@ public function select($columnArray)
         if (empty($columnArray)) {
             throw new PropelException('You must ask for at least one column');
         }
+        $this->isSelfSelected = true;
+        if ($this->formatter === null) {
+            $this->setFormatter('\Propel\Runtime\Formatter\SimpleArrayFormatter');
+        }
 
         if ($columnArray === '*') {
             $columnArray = [];
-            foreach (call_user_func([$this->modelTableMapName, 'getFieldNames'], TableMap::TYPE_PHPNAME) as $column) {
-                $columnArray[] = $this->modelName . '.' . $column;
+            foreach ($this->getTableMap()->getColumns() as $columnMap) {
+                $columnArray[] = $this->modelName . '.' . $columnMap->getPhpName();
             }
         }
+        if (!is_array($columnArray)) {
+            $columnArray = [$columnArray];
+        }
 
-        $this->select = $columnArray;
+        $this->selectColumns = [];
 
-        return $this;
-    }
+        foreach ($columnArray as $columnName) {
+            if (array_key_exists($columnName, $this->asColumns)) {
+                continue;
+            }
+            [$columnMap, $realColumnName] = $this->getColumnFromName($columnName);
+            if ($realColumnName === null) {
+                throw new PropelException("Cannot find selected column '$columnName'");
+            }
+            // always put quotes around the columnName to be safe, we strip them in the formatter
+            $this->addAsColumn('"' . $columnName . '"', $realColumnName);
+        }
 
-    /**
-     * Retrieves the columns defined by a previous call to select().
-     *
-     * @see select()
-     *
-     * @return array|string A list of column names (e.g. array('Title', 'Category.Name', 'c.Content')) or a single column name (e.g. 'Name')
-     */
-    public function getSelect()
```

Comment:
This function is also part of the public interface of this library, and we shouldn't remove it haphazardly.